Python/Product/PythonTools/PythonTools/Editor/Formatting/PythonFormatCommandHandler.cs

```diff
@@ -195,100 +177,47 @@ string[] extraArgs
-                MessageBoxButton.YesNo,
-                MessageBoxImage.Question
-            );
-
-            if (result == MessageBoxResult.Yes) {
+            var message = Strings.InstallFormatterPrompt.FormatUI(formatter.Package, factory.Configuration.Description);
+            if (ShowYesNoPrompt(message)) {
                 try {
-                    return await pm.InstallAsync(
-                        PackageSpec.FromArguments(formatter.PackageSpec),
-                        new VsPackageManagerUI(_site),
-                        CancellationToken.None
-                    );
+                    return await pm.InstallAsync(PackageSpec.FromArguments(formatter.Package), new VsPackageManagerUI(_site), CancellationToken.None);
                 } catch (Exception ex) when (!ex.IsCriticalException()) {
-                    return false;
+                    ShowErrorMessage(Strings.ErrorUnableToInstallFormatter.FormatUI(ex.Message));
                 }
             }
-
-            return null;
+            return false;
         }
 
-        private bool GetConfiguration(
-            ITextDocument textDoc,
-            out IPythonFormatter formatter,
-            out IPythonInterpreterFactory factory,
-            out string[] extraArgs
-        ) {
+        private bool GetConfiguration(out IPythonFormatter formatter, out string[] extraArgs) {
             formatter = null;
-            factory = null;
             extraArgs = new string[0];
 
-            const string defaultFormatterId = "black";
-            string formatterId = null;
-
-            if (_workspaceContextProvider.Workspace != null) {
-                // Workspace file
-                formatterId = _workspaceContextProvider.Workspace.GetStringProperty(PythonConstants.FormatterSetting);
-                factory = _workspaceContextProvider.Workspace.CurrentFactory;
-            } else {
-                var project = _site.GetProjectContainingFile(textDoc.FilePath);
-                if (project != null) {
-                    // Project file
-                    formatterId = project.GetProjectProperty(PythonConstants.FormatterSetting);
-                    factory = project.ActiveInterpreter;
-                } else {
-                    // Loose file
-                    formatterId = defaultFormatterId; // TODO: use a global option?
-                    factory = _optionsService.DefaultInterpreter;
-                }
-            }
-
-            // TODO: if not configured, tell the user? for now use default
-            if (string.IsNullOrEmpty(formatterId)) {
-                formatterId = defaultFormatterId;
-            }
-
-            if (!string.IsNullOrEmpty(formatterId)) {
-                formatter = _formattingProviders.SingleOrDefault(p => string.Compare(p.Value.Identifier, formatterId, StringComparison.OrdinalIgnoreCase) == 0)?.Value;
-            }
+            var formatterId = _site.GetPythonToolsService().FormattingOptions.Formatter;
```

Comment:
Please let's keep doing the workspace/project check first, then if not set, get the global default.
That's still useful even though it requires manual editing of json/pyproj at this time.

Reply:
Question is - what might store the option there. The file is JSON inside `.vs` hidden folder. So in order to have something there, we'd somehow need to store information there. Since there is no code that would be storing data, I decided not to read it. Technically in VS formatting is in `Tools | Options | Text Editor | Python | Formatting` (as it is with other languages) so setting is always global - otherwise each time use opened `Tools | Options` the setting would come different depending what workspace or project is active.

Reply:
While there is no UI for editing the workspace settings in PythonSettings.json, we still have settings. The experience is bad compared to VS code which has a UI on top of settings, but that's what we have so far. User is expected to hand edit that file right now. For things like unit test framework, search paths, formatter, and interpreter (that last one has a UI -- the python toolbar writes that one).

Having the ability to set the formatter per workspace/project is important. Hopefully we can invest in a UI later, but until then, it's better if the user can edit the setting in the json editor than not being to control it other than global setting.

Reply:
Oh, I see, it is `hidden` but documented file... :-)

Reply:
Correct. One of the reasons workspace experience is currently inferior to projects.

Example of documentation: https://docs.microsoft.com/en-us/visualstudio/python/unit-testing-python-in-visual-studio?view=vs-2019#configure-testing-for-python-without-a-project

Reply:
OK, restored it.